h/auth/util.py

```diff
@@ -117,6 +118,42 @@ def default_authority(request):
     return text_type(request.registry.settings.get('h.authority', request.domain))
 
 
+def client_authority(request):
+    """
+    Return the authority associated with an authenticated auth_client or None
+
+    Once a request with an auth_client is authenticated, a principal is set
+    indicating the auth_client's verified authority
+
+    see :func:`~h.auth.util.principals_for_auth_client` for more details on
+    principals applied when auth_clients are authenticated
+
+    :rtype: str or None
+    """
+    for principal in request.effective_principals:
+        match = re.match(r"^authority:(.+)$", principal)
+        if match and match.group(1):
+            return match.group(1)
+
+    return None
+
+
+def authority(request):
+    """
+    Return active authority for the current request.
+
+    Return the client_authority, if there is one, or the default authority
+
+    :rtype: str
+    """
+
+    active_authority = client_authority(request)
+    if active_authority is not None:
+        return active_authority
+
+    return default_authority(request)
```

Comment:
I recognize that the coding style in this function is verbose, but I wanted it to be _super clear_ because it's important.